policy/modules/services/incus.te

```diff
@@ -0,0 +1,206 @@
+# scontext=system_u:system_r:incusd_t:s0
+# tcontext=system_u:object_r:sysctl_vm_overcommit_t:s0 tclass=file permissive=0
+kernel_read_vm_overcommit_sysctl(incusd_t)
+
+# for bridge management
+dev_create_sysfs_files(incusd_t)
+dev_write_sysfs(incusd_t)
+
+# incusd won't start without access to /dev/urandom
+dev_read_urand(incusd_t)
+
+# mounton access to /proc/sys/kernel/random/boot_id
+kernel_mounton_kernel_sysctl_files(incusd_t)
+
+# read /etc/machine-id
+files_read_etc_runtime_files(incusd_t)
+
+# incus apparmor support wants to handle /sys/kernel/tracing
+fs_dontaudit_getattr_tracefs(incusd_t)
+fs_dontaudit_getattr_tracefs_dirs(incusd_t)
+fs_dontaudit_remount_tracefs(incusd_t)
+
+# reason for access not (yet?) looked up in the source code
+# but everything seems to work fine so far
+fs_dontaudit_getattr_configfs(incusd_t)
+fs_dontaudit_remount_configfs(incusd_t)
+
+# required for fanotify on /dev/zvol/zdata/incus/virtual-machines/.#vm1.block...
+dev_watch_dev_fs(incusd_t)
+
+# required so that incus will get the VM PID
+# and make 'incus info VM' work
+# Error: stat /proc/628420: permission denied
+qemu_read_state(incusd_t)
+
+tunable_policy(`incus_lxc_legacy_mounts',`
+	fs_getattr_binfmt_misc_fs(incusd_t)
+
+	kernel_getattr_debugfs(incusd_t)
+	kernel_search_debugfs(incusd_t)
+	kernel_remount_debugfs(incusd_t)
+
+	fs_getattr_efivarfs(incusd_t)
+	fs_remount_efivarfs(incusd_t)
+
+	fs_getattr_pstorefs(incusd_t)
+	fs_remount_pstorefs(incusd_t)
+',`
+	fs_dontaudit_getattr_binfmt_misc_fs(incusd_t)
+
+	fs_dontaudit_getattr_efivarfs(incusd_t)
+	fs_dontaudit_remount_efivarfs(incusd_t)
+
+	fs_dontaudit_getattr_pstorefs(incusd_t)
+	fs_dontaudit_remount_pstorefs(incusd_t)
+
+	kernel_dontaudit_getattr_debugfs(incusd_t)
+	kernel_dontaudit_search_debugfs(incusd_t)
+	kernel_dontaudit_remount_debugfs(incusd_t)
+')
+
+# incus idmap support: checks whether the kernel supports VFS v3 fscaps
+container_execute_engine_tmp_files(incusd_t)
+
+# required for QEMU feature check, won't start otherwise
+files_rw_generic_tmp_sockets(incusd_t)
+files_dontaudit_unlink_generic_tmp_sockets(incusd_t)
+
+qemu_stream_connect(incusd_t)
+qemu_kill(incusd_t)
+dev_getattr_vhost_dev(incusd_t)
+
+# these are required for incusd to check and execute qemu
+qemu_domtrans(incusd_t)
+qemu_exec(incusd_t)
+
+# incusd needs support to label VM files before launching
+# until then, VM files will unfortunately be unlabeled
+kernel_manage_unlabeled_dirs(incusd_t)
+kernel_manage_unlabeled_files(incusd_t)
+kernel_manage_unlabeled_symlinks(incusd_t)
+kernel_mounton_unlabeled_dirs(incusd_t)
+files_search_default(incusd_t)
+
+# name="stub-resolv.conf"
+fs_read_tmpfs_files(incusd_t)
+
+########################################
+#
+# incus lxc container local policy
+
+allow incus_container_domain self:cap_userns { chown dac_override dac_read_search fowner fsetid kill net_admin net_bind_service net_raw setgid setpcap setuid sys_admin sys_boot sys_chroot sys_ptrace };
```

Comment:
`dac_override` can probably be dontaudit'd if it has `dac_read_search`, may be worth testing without `dac_override`.

Reply:
removed the dac_override